Controllers/SongController.cs

```diff
@@ -85,6 +90,51 @@ public async Task<ActionResult<Song>> RefreshHash(string hash)
             return song;
         }
 
+        [HttpGet("~/map/modinterface/{hash}")]
+        public async Task<ActionResult<List<DiffModResponse>>> GetModSongInfos(string hash)
+        {
+            ICollection<DifficultyDescription>? diffs = await _context.Songs
+                .Where(el => el.Hash == hash)
+                .Select(song => song.Difficulties)
+                .FirstOrDefaultAsync();
+
+            if (diffs is null)
+            {
+                return NotFound();
+            }
+
+            List<DiffModResponse> result = new();
+
+            foreach(DifficultyDescription? diff in diffs)
+            {
+                if(diff is not null)
+                {
+                    result.Add(new()
+                    {
+                        DifficultyName = diff.DifficultyName,
+                        ModeName = diff.ModeName,
+                        Stars = diff.Stars,
+                        Nominated = diff.Nominated,
+                        Qualified = diff.Qualified,
+                        Ranked = diff.Ranked,
+                        Type = diff.Type,
+                        Votes = await _context.RankVotings
```

Comment:
`Score` has a `RankVoting` in it. So you probably can fetch everything you need in one query from the leaderboards. Smth like 
```
_context.Leaderboards.Select(lb => 
new DiffModResponse { DifficultyName = lb.Difficulty.DifficultyName ... yada yada ... Votes = lb.Scores.Select(s => s.RankVoting.Rankability) })
```

Reply:
I'm a little worried that it would then download all the scores locally and not just the RankVotings. But to judge something like that I lack the knowledge about EF ^^ (I dont even know the transfer speed between the BL Server and the SQL Server)

Reply:
Also, I just noticed that this does not work for songs (or diffs) that do not have a score yet. For these I get no stars and also not the ranked status.

Or do you have an idea how it could still work?

Reply:
Fetching from songs will not work either. You need to do the fallback like in the "/map/hash/{hash}"

Reply:
Implemented the Fallback without the Leaderboard query (as this would still require scores on the diffs/modes :( )

Reply:
If you will select-out the scores - they will not be transmitted

Reply:
Sorry if I wasn't clear. I meant, that I get no ranked Status/Stars for songs, that nobody played before, as there is no leaderboard existing (yet).

I could create one if no leaderboard exists (like in LeaderboardController.GetByHash), if you want that. But in 99% of the requests, it probably already exists. So that's probably still better than my two requests from before ^^